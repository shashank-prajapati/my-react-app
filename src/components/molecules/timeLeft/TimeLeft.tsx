import React from 'react';
import Typography from '@mui/material/Typography';
import Box, { BoxProps } from '@mui/material/Box';
import ClockIcon from '../../atoms/ClockIcon/ClockIcon';

interface Props extends BoxProps{
  time:string;
}

function TimeLeft(props: Props) {
  const{sx,time,...rest} = props;
return (<div>
        <Box sx={{
            display:'flex',
            alignItems:'center',
            ...sx    
        }}>
            <ClockIcon/>
            <Typography variant='caption3' color='extraTextColor.main' sx={{ml:2}}>{time}-minute read</Typography>
        </Box>       
  </div>);
}

export default TimeLeft;
