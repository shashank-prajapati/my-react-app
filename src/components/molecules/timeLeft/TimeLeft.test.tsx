import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import TimeLeft from "./TimeLeft";

describe("Testing Footer List", () => {
  it("should match the Time Left snapshot", () => {
    const { container } = render(<TimeLeft time="13" />);
    expect(container).toMatchSnapshot();
  });
});
