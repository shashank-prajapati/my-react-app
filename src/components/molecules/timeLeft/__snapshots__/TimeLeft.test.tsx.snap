// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer List should match the Time Left snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-70qvj9"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
        focusable="false"
      >
        clockIcon.svg
      </svg>
      <span
        class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
      >
        13
        -minute read
      </span>
    </div>
  </div>
</div>
`;
